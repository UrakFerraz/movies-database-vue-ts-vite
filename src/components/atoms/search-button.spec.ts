import { mount } from '@vue/test-utils'
import SearchButton from './search-button.vue'

const wrapper = mount(SearchButton)

describe('deve ser o componente button', () => {
	it('deve ser um componente button', () => {
		expect(wrapper).toBeTruthy()
		expect(wrapper.html()).toBeDefined
		const button = wrapper.find('button')
		expect(button.exists()).toBe(true)
		const svg = wrapper.find('svg')
		expect(svg.exists()).toBe(true)
		expect(wrapper.html()).toMatchSnapshot()
	})
})