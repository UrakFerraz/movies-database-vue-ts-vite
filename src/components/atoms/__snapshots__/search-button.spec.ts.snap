// Vitest Snapshot v1

exports[`deve ser o componente buttom > deve ser um componente buttom 1`] = `
"<button class=\\"search-button\\" data-v-68a0f20a=\\"\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\" class=\\"search-button__svg\\" data-v-68a0f20a=\\"\\">
    <path fill=\\"currentColor\\" d=\\"M500.3 443.7l-119.7-119.7c27.22-40.41 40.65-90.9 33.46-144.7C401.8 87.79 326.8 13.32 235.2 1.723C99.01-15.51-15.51 99.01 1.724 235.2c11.6 91.64 86.08 166.7 177.6 178.9c53.8 7.189 104.3-6.236 144.7-33.46l119.7 119.7c15.62 15.62 40.95 15.62 56.57 0C515.9 484.7 515.9 459.3 500.3 443.7zM79.1 208c0-70.58 57.42-128 128-128s128 57.42 128 128c0 70.58-57.42 128-128 128S79.1 278.6 79.1 208z\\" data-v-68a0f20a=\\"\\"></path>
  </svg></button>"
`;

exports[`deve ser o componente button > deve ser um componente button 1`] = `
"<button class=\\"search-button\\" data-v-68a0f20a=\\"\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\" class=\\"search-button__svg\\" data-v-68a0f20a=\\"\\">
    <path fill=\\"currentColor\\" d=\\"M500.3 443.7l-119.7-119.7c27.22-40.41 40.65-90.9 33.46-144.7C401.8 87.79 326.8 13.32 235.2 1.723C99.01-15.51-15.51 99.01 1.724 235.2c11.6 91.64 86.08 166.7 177.6 178.9c53.8 7.189 104.3-6.236 144.7-33.46l119.7 119.7c15.62 15.62 40.95 15.62 56.57 0C515.9 484.7 515.9 459.3 500.3 443.7zM79.1 208c0-70.58 57.42-128 128-128s128 57.42 128 128c0 70.58-57.42 128-128 128S79.1 278.6 79.1 208z\\" data-v-68a0f20a=\\"\\"></path>
  </svg></button>"
`;
